### ISSUE TITLE
Symmetric Difference Algorithm not Testing for Symmetric Difference

### ISSUE THREAD

**Issue by angkiki** [freeCodeCamp/freeCodeCamp#12598]
While I understand that the challenge can come up with all sorts of rules, it is quite confusing since it really isn't testing for the _symmetric difference_ between all the given arrays.

Example: 
sym([1, 2, 5], [2, 3, 5], [3, 4, 5]) should return [1, 4, 5]

That isn't _"correct"_ since the actual symmetric difference among the 3 arrays should be [1,4].

On the challenge help, a link to a [youtube video](https://www.youtube.com/watch?v=PxffSUQRkG4) was provided. 

In the video, the following example was given:
A = [1,2,3,4]
B = [3,4,5,6]
C = [2,4,6,7]
And the symmetric difference between the 3 sets is [1,2,5,7]. Comparing with the answers expected from the challenge, which in this instance would be [1,4,5,7].

Once again, I know that the challenges can have its own rules to it. I just thought that I should point out this fact.

**Comment by erictleung** [comment]
@angkiki thanks for the issue. Agreed, this isn't a *true* symmetric difference, but the instructions are clear as to what to do when there is more than one array.

> For every additional symmetric difference you take (say on a set D = {2, 3}), you should get the set with elements which are in either of the two the sets but not both (C △ D = {1, 4} △ {2, 3} = {1, 2, 3, 4}).

So I'm not sure what else is needed. Explaining within the challenge that this isn't a *true* symmetric difference among input of three or more arrays would be confusing. Thoughts?

**Comment by angkiki** [comment]
Once again i do apologise for "nit picking". I will admit that through the list of expected answers one can infer what is expected. 

If i were to offer an input, maybe state that "each additional array is compared against the resulted symmetric difference". It seems more explicit with that sentence.  (Or it may just be my poor interpretation of the example listed)

**Comment by no-stack-dub-sack** [comment]
@erictleung I'm with @angkiki on this one, I happen to think the instructions for this challenge (especially for those that might not be familiar with this type of notation, i.e. C △ D = {1, 4} △ {2, 3} = {1, 2, 3, 4} ) are quite convoluted and hard to follow. And while you're right, the second part is (kind of) clear on what it's asking you to do, the first part simply states:

> Create a function that takes two or more arrays and returns an array of the symmetric difference (△ or ⊕) of the provided arrays.

Which I guess is technically incorrect. If someone happens to not read any further, and we know this happens often, I can see where the confusion could come in.

**Comment by erictleung** [comment]
@no-stack-dub-sack sounds reasonable. How about this change to the first sentence? (My additions in bold.)

> Create a function that takes two or more arrays and returns an array of the **iterative** symmetric difference (△ or ⊕) of the provided arrays. **In other words, start with the first array and keep on taking symmetric differences of each the following arrays after it.**

**Comment by no-stack-dub-sack** [comment]
@erictleung That sounds good. That, in addition to looking at the test cases should provide clearer direction I think.

**Comment by Syntactual** [comment]
i could take this ; this would be my first time

**Comment by Syntactual** [comment]
can't find this challenge in any of the challenges; which file is this question located?

**Comment by angkiki** [comment]
@no-stack-dub-sack i would love to but i am actually really unfamiliar with working with git...

**Comment by ChrisD1986** [comment]
Is our problem not the ontological understanding of symmetric
differentiation.
